src/ray/raylet/node_manager.cc

```diff
@@ -2047,6 +2047,12 @@ void NodeManager::ProcessSetResourceRequest(
 bool NodeManager::PrepareBundle(
     std::unordered_map<ClientID, SchedulingResources> &resource_map,
     const BundleSpecification &bundle_spec) {
+  // We will first delete the existing bundle to ensure idempotent.
+  // The reason why we do this is: after GCS restarts, placement group can be rescheduled directly
+  // without rolling back the operations performed before the restart.
+  const auto &bundle_id = bundle_spec.BundleId();
```

Comment:
Actually, this shouldn't be enough because if there was a bundle in prepare state, it already locked resources. To avoid it, we should "remove" bundles that are in prepare state when gcs restarts is detected. This is specified in the doc as well. 

```
If GCS server shutdown is detected, every raylet just cancels prepare requests.
This will make all raylets the correct state for PENDING creation.
```
https://docs.google.com/document/d/1tPMs2W-557hPmln5xnycK85vKTxJujZIsjq9Rsw9OzY/edit#

Reply:
I will fix it, thanks.